net/FlatBuffers/ByteBuffer.cs

```diff
@@ -523,38 +509,56 @@ public unsafe void PutLong(int offset, long value)
         public unsafe void PutUlong(int offset, ulong value)
         {
             AssertOffsetAndLength(offset, sizeof(ulong));
-            byte* ptr = _buffer.Buffer;
-            *(ulong*)(ptr + offset) = BitConverter.IsLittleEndian
-                ? value
-                : ReverseBytes(value);
+#if ENABLE_SPAN_T
+            Span<byte> span = _buffer.GetSpan(offset);
+            BinaryPrimitives.WriteUInt64LittleEndian(span, value);
+#else
+            fixed (byte* ptr = _buffer.Buffer)
+            {
+                *(ulong*)(ptr + offset) = BitConverter.IsLittleEndian
+                    ? value
+                    : ReverseBytes(value);
+            }
+#endif
         }
 
         public unsafe void PutFloat(int offset, float value)
         {
             AssertOffsetAndLength(offset, sizeof(float));
-            byte* ptr = _buffer.Buffer;
-            if (BitConverter.IsLittleEndian)
-            {
-                *(float*)(ptr + offset) = value;
-            }
-            else
+#if ENABLE_SPAN_T
+            fixed (byte* ptr = &MemoryMarshal.GetReference(_buffer.GetSpan()))
+#else
+            fixed (byte* ptr = _buffer.Buffer)
+#endif
             {
-                *(uint*)(ptr + offset) = ReverseBytes(*(uint*)(&value));
+                if (BitConverter.IsLittleEndian)
+                {
+                    *(float*)(ptr + offset) = value;
+                }
+                else
+                {
+                    *(uint*)(ptr + offset) = ReverseBytes(*(uint*)(&value));
+                }
```

Comment:
Can this instead be:
```C#
Span<byte> span = _buffer.GetSpan(offset);
BinaryPrimitives.WriteInt32LittleEndian(span, BitConverter.SingleToInt32Bits(value));
```
?

Reply:
Unfortunately no.  :(

```
Severity	Code	Description	Project	File	Line	Suppression State
Error	CS0117	'BitConverter' does not contain a definition for 'SingleToInt32Bits'	Apache.Arrow(netstandard1.3)	F:\git\arrow\csharp\src\Apache.Arrow\Flatbuf\FlatBuffers\ByteBuffer.cs	514	Active
Error	CS0117	'BitConverter' does not contain a definition for 'Int32BitsToSingle'	Apache.Arrow(netstandard1.3)	F:\git\arrow\csharp\src\Apache.Arrow\Flatbuf\FlatBuffers\ByteBuffer.cs	717	Active
```

The Int32 and float APIs are .NET Core 2.0 only. https://apisof.net/catalog/System.BitConverter.Int32BitsToSingle(Int32)

The Int64 and double APIs have been there since .NET Framework 1.0. https://apisof.net/catalog/System.BitConverter.Int64BitsToDouble(Int64)

Reply:
> The Int32 and float APIs are .NET Core 2.0 only. 

Ah 😦 Ok. Does raise the question of whether (eventually, obviously not part of this PR) there should be a netcoreapp build of the library in addition to netstandard2.0.

That said, the implementation is just `*((int*)&value)`, so it can be done without the API.

Reply:
> Does raise the question of whether (eventually, obviously not part of this PR) there should be a netcoreapp build of the library in addition to netstandard2.0.

The way the Flat Buffers code "ships" (as far as I can tell) is that users copy this code into their projects. So there isn't really a "build of the library". So to make `netcoreapp` vs. `netstandard` code here would involve more `#if` defines, and then the users would have to set the appropriate defines.